src/custom/state/orders/updaters/UnfillableOrdersUpdater.ts

```diff
@@ -56,7 +57,8 @@ async function _getOrderPrice(chainId: ChainId, order: Order, strategy: GpPriceS
     quoteToken,
     fromDecimals: order.inputToken.decimals,
     toDecimals: order.outputToken.decimals,
-    validTo: timestamp(order.validTo),
+    validTo:
+      order.class === 'limit' ? Math.round((Date.now() + PRICE_QUOTE_VALID_TO_TIME) / 1000) : timestamp(order.validTo),
```

Comment:
Was this somehow part of the problem? Maybe it would be nice to mention in the description high level the changes you applied to hint what is the solution. 

Also, should we refactor this with:
https://github.com/cowprotocol/cowswap/pull/1525/files#diff-3e3c40ca973eab122a07ee5968d5d8a3834c6e8ea9e6d9e21294cc5bad62101dR66

Reply:
Yes, it's a check to avoid ExcessiveValidTo. Added comment.

> Also, should we refactor this with

Unfortunately, I don't fully understand that code, maybe